lib/crypto.js

```diff
@@ -276,13 +276,18 @@ Sign.prototype._write = function _write(chunk, encoding, callback) {
 
 Sign.prototype.update = Hash.prototype.update;
 
-Sign.prototype.sign = function sign(options, encoding) {
-  if (!options)
+Sign.prototype.sign = function sign(privateKey, encoding, options) {
+  if (!privateKey)
     throw new Error('No key provided to sign');
 
-  var key = options.key || options;
-  var passphrase = options.passphrase || null;
-  var ret = this._handle.sign(toBuf(key), null, passphrase);
+  if (typeof encoding == 'object' && !options) {
+    options = encoding;
+    encoding = null;
+  }
+
+  var key = privateKey.key || privateKey;
+  var passphrase = privateKey.passphrase || null;
```

Comment:
Is this still relevant when changing the `options` object to be the first argument, such that the first argument is either the `privateKey` itself or an `options` object with properties `key` and optionally `passphrase`, `padding` and/or `saltLength`?